Makefile

```diff
@@ -27,8 +30,81 @@ DOCKER_GO_BUILD := mkdir -p .go-pkg-cache && \
                               -w /go/src/github.com/$(PACKAGE_NAME) \
                               $(CALICO_BUILD)
 
-VENDOR_REMADE := false
+# Create a list of files upon which the generated file depends, skip the generated file itself
+UPGRADE_SRCS := $(filter-out ./lib/upgrade/migrator/clients/v1/k8s/custom/zz_generated.deepcopy.go, \
+                             $(wildcard ./lib/upgrade/migrator/clients/v1/k8s/custom/*.go))
+
+# Create a list of files upon which the generated file depends, skip the generated file itself
+APIS_SRCS := $(filter-out ./lib/apis/v3/zz_generated.deepcopy.go, $(wildcard ./lib/apis/v3/*.go))
+
+.PHONY: clean
+## Removes all .coverprofile files, the vendor dir, and .go-pkg-cache
+clean:
+	find . -name '*.coverprofile' -type f -delete
+	rm -rf vendor .go-pkg-cache
+	rm -rf $(BINDIR)
+
+.PHONY: clean-gen-files
+## Convenience target for devs to remove generated files and related utilities
+clean-gen-files:
+	rm -f $(BINDIR)/deepcopy-gen
+	rm -f ./lib/apis/v3/zz_generated.deepcopy.go
+	rm -f ./lib/upgrade/migrator/clients/v1/k8s/custom/zz_generated.deepcopy.go
+
+###############################################################################
+# Building the binary
+###############################################################################
+.PHONY: update-vendor
+# Update the pins in glide.lock to reflect the updated glide.yaml.
+# Note: no dependency on glide.yaml so we don't automatically update glide.lock without
+# developer intervention.
+update-vendor glide.lock:
```

Comment:
I don't remember seeing an `update-vendor` target in the other PRs - do we want / need this?

Reply:
As discussed, we think not.